### PR TITLE
Jump to top of page on navigation

### LINKED ISSUE #162: [Bug]: Some form of Scroll to top needs implementing
## Description

When navigating the site internally scroll position is remembered. We need to implement some sort of scroll to top function on page load.

## Steps to reproduce

1. Visit the site, and scroll down the page
2. Click on an internal link

## Platform (if relevant)

Chrome, on a mac

### PR DESCRIPTION
**Pull request by ivan-kocienski-gfsc**
Fixes #162 

## Description

When clicking a link the browser window is reset to the top of the new page.

Also all the primary headers are rendered through one point so styling is consistent across all pages.

@geeksforsocialchange/developers

### CODE REVIEW

File: src/Theme/PageTemplate.elm [hunk 1]
```diff
@@ -14,6 +14,7 @@ import Theme.HeaderTemplate as HeaderTemplate
 view : Model -> Html Msg -> Html Msg
 view model content =
     div []
+        -- id "focus-target" ]
```
Comment (r-ferrier) [inline src/Theme/PageTemplate.elm:17]: needs removing?

File: src/Theme/Global.elm [hunk 2]
```diff
@@ -178,6 +180,16 @@ globalStyles =
...
 
 
+primaryHeader : String -> Html Msg
+primaryHeader title =
+    h1 [ css [ primaryHeaderStyle ], id "focus-target" ] [ text title ]
+
+
+primaryHeaderStyle : Style
+primaryHeaderStyle =
+    fontFamilies [ "Ludicrous", "serif" ]
```
Comment (r-ferrier) [inline src/Theme/Global.elm:210]: I don't think this is right - I think this font used to be used on the guide and story pages but not any of the index pages, and it looks like it's not in use on any of the headers any more https://miro.com/app/board/uXjVPyf-Msw=/

Comment (r-ferrier) [reply src/Theme/Global.elm:210]: I'm also wondering if this style is necessary - we already apply styling to all h1 tags globally, so I think you could just remove this

File: src/Theme/Global.elm [hunk 3]
```diff
@@ -178,6 +180,16 @@ globalStyles =
 -- Helpers
 
 
+primaryHeader : String -> Html Msg
+primaryHeader title =
+    h1 [ css [ primaryHeaderStyle ], id "focus-target" ] [ text title ]
```
Comment (r-ferrier) [inline src/Theme/Global.elm:205]: This is a good idea! It looks like you're just missing the `tabindex -1` attribute Katja had in her example

**Review by r-ferrier — CHANGES_REQUESTED** [review]
Looks good, just a couple of changes requested

File: src/Theme/Global.elm (hunk 2 above)
Comment (ivan-kocienski-gfsc) [reply src/Theme/Global.elm:210]: That feels bad to me: having an inconsistent style on pages.

Comment (r-ferrier) [reply src/Theme/Global.elm:210]: What do you mean? All the pages have the same style now, they just used to have a different one - if you have. alook at the miro you can see what I mean! They just don't have this font - they should just have the styles being applied in theme.global to the h1 tags automatically

Comment (ivan-kocienski-gfsc) [reply src/Theme/Global.elm:210]: no problem, removed local styling

File: src/Theme/Global.elm (hunk 3 above)
Comment (ivan-kocienski-gfsc) [reply src/Theme/Global.elm:205]: added `tabindex`

File: src/Page/Guides/View.elm [hunk 4]
```diff
@@ -37,16 +37,17 @@ view model =
...
+        headerText : String
+        headerText =
+            if String.length model.query > 0 then
+                t <| GuidesTitleFiltered (String.fromInt <| List.length model.search) model.query
+
+            else
+                t GuidesTitle
     in
     div [ css [ centerContent ] ]
-        [ h1 [ attribute "aria-live" "alert" ]
```
Comment (r-ferrier) [inline src/Page/Guides/View.elm:42]: I think we're losing this attribute we need for screen readers from this page - maybe you could create another component, `searchResultPrimaryText` or something like that to be able to preserve this?

File: src/Theme/PageTemplate.elm [hunk 5]
```diff
@@ -3,7 +3,7 @@ module Theme.PageTemplate exposing (view)
 import CookieBanner
 import Css exposing (Style, backgroundColor, batch, border3, hidden, overflowX, pct, rem, solid, width)
 import Html.Styled exposing (Html, div, main_)
-import Html.Styled.Attributes exposing (css)
+import Html.Styled.Attributes exposing (css, id)
```
Comment (r-ferrier) [inline src/Theme/PageTemplate.elm:6]: remove?

**Review by r-ferrier — CHANGES_REQUESTED** [review]
Unfortunately one of the h1s is now slightly different to the others and needs an additional attribute for screen readers - have made a suggestion about how to possibly handle this but up to you!

File: src/Theme/PageTemplate.elm (hunk 5 above)
Comment (ivan-kocienski-gfsc) [reply src/Theme/PageTemplate.elm]: removed

**Comment by r-ferrier** [comment]
LGTM!

File: src/Page/Guides/View.elm [hunk 6]
```diff
@@ -37,16 +37,17 @@ view model =
 
                     Success list ->
                         List.concat [ Page.Guides.Data.teaserListFromGuideDict model.language model.content.guides, list ]
+
+        headerText : String
+        headerText =
+            if String.length model.query > 0 then
+                t <| GuidesTitleFiltered (String.fromInt <| List.length model.search) model.query
```
Comment (katjam) [inline src/Page/Guides/View.elm:44]: We aren't using this sytax elsewhere. It's probably best for readabilitty if we stick with () for this direction of application when there are one or 2 functions to apply and |> where there is a longer chain of functions.

**Pull request merged**